src/clover/optimisation/appraisal.py

```diff
@@ -362,14 +416,26 @@ def _simulation_financial_appraisal(  # pylint: disable=too-many-locals
         start_year=system_details.start_year,
         end_year=system_details.end_year,
     )
-    grid_costs = finance.expenditure(
-        ImpactingComponent.GRID,
-        finance_inputs,
-        simulation_results[ColumnHeader.GRID_ENERGY.value],
-        logger,
-        start_year=system_details.start_year,
-        end_year=system_details.end_year,
-    )
+    # This function will need:
+    #   - to know how much energy was used, from each grid
+
+    #APPRAISAL TO CHANGE: 
+
+    for grid_name in scenario.grid_types:
```

Comment:
Okay regarding this point, and for me to understand further. When you say, .costs or .tiers or .anything will return the dictionary it means, it will return the costs dictionary that were set previously in fileparser? Like my question here, how is clover knowing what costs to read and how can I direct it toward the cost I want.

Regarding the dictionary reading, so we want to take this dictionary and extract the costs out of it so we can assign a specific cost for the calculation here... not the entire dictionary? Can you explain further (maybe attached with a small example)... These are confusing me.

Reply:
On the first point, I suggest playing around with it: use
```python
import pdb

pdb.set_trace()
```
to explore what the classes do, how they work, and how the information is available at this point.

You should be directing CLOVER to the specific costs using your `which_tier_am_i_in` function :smile:

Reply:
Hi @BenWinchester , I tried putting 
```
import pdb
pdb.set_trace()
```
in the tier function, nothing is showing and I can't find what classes like you mentioned...

Reply:
Hi @paulharfouche , the `pdb` debugger only works so long as the code is reached. I.E., it basically stops python as it's executing the code at the point that you write it. If the code isn't reaching this point, then it won't run.

Make sure you're using the `--analyse` flag when you run CLOVER :smile: